sogs/model/message.py

```diff
@@ -42,6 +42,27 @@ def __init__(self, row=None, *, sender=None, recip=None, data=None):
         assert row is not None
         self._row = row
 
+    @staticmethod
+    def delete_all(*, recip=None, sender=None):
+        """Delete all messages sent to a user or from a user.
+        returns the number of rows affected.
+        """
+        if sender and recip:
+            raise ValueError("delete_all(): cannot provide both sender and recipient")
```

Comment:
Also needs to fail if you pass in neither (because otherwise this query is going to delete all DMs of all users if two `None`s get passed in here).